cirq/protocols/json_serialization.py

```diff
@@ -702,3 +703,42 @@ def obj_hook(x):
             return json.load(file, object_hook=obj_hook)
 
     return json.load(cast(IO, file_or_fn), object_hook=obj_hook)
+
+
+def to_gzip(
+    obj: Any,
+    file_or_fn: Union[None, IO, pathlib.Path, str] = None,
+    *,
+    indent: int = 2,
+    cls: Type[json.JSONEncoder] = CirqEncoder,
+) -> Optional[bytes]:
+    json_str = to_json(obj, indent=indent, cls=cls)
+    gzip_data = gzip.compress(bytes(json_str, encoding='utf-8'))  # type: ignore
```

Comment:
do you know the difference between `bytes(str)` and `str.encode(...)`

Reply:
It might be better to use the gzip file-like objects so it can stream to files, which might make a difference for large objects

Reply:
`bytes(str)` and `str.encode` have identical behavior: [StackOverflow answer](https://stackoverflow.com/questions/14472650/python-3-encode-decode-vs-bytes-str)

Updated to use gzip files where possible.

Reply:
Thanks for the link! Learned something new [I guess I was stuck in python 2 compatibility mode]